src/spandrel/architectures/MMRealSR/__init__.py

```diff
@@ -0,0 +1,172 @@
+from __future__ import annotations
+
+from typing import Literal
+
+from ...__helpers.model_descriptor import (
+    ImageModelDescriptor,
+    SizeRequirements,
+    StateDict,
+)
+from ..__arch_helpers.state import get_seq_len
+from .arch.mmrealsr_arch import MMRRDBNet_test as MMRealSR
+
+
+def _get_in_ch_and_scale(combined: int, num_out_ch: int) -> tuple[int, int]:
+    # num_in_ch and scale are linked in the original code like this:
+    #   if scale == 2:
+    #       num_in_ch = num_in_ch * 4
+    #   elif scale == 1:
+    #       num_in_ch = num_in_ch * 16
+    # It is not possible to read the original values for num_in_ch and scale, so we
+    # have to deduce them from the combined value. Luckily, only scale=1,2,4 are supported.
+    # We'll also assume that num_in_ch == num_out_ch is very likely
+    if num_out_ch * 1 == combined:
+        num_in_ch = num_out_ch
+    elif num_out_ch * 4 == combined:
+        num_in_ch = num_out_ch
+    elif num_out_ch * 16 == combined:
+        num_in_ch = num_out_ch
+    elif combined % 3 == 0:
+        num_in_ch = 3
+    elif combined % 16 == 0 and combined > 16:
+        num_in_ch = combined // 16
+    elif combined % 4 != 0:
+        num_in_ch = combined
+    else:
+        # it's impossible to differentiate between 2x RGBA vs 1x grayscale and
+        # 4x RGBA vs 2x grayscale, so we assume that 2x models are unlikely
+        if combined == 16:
+            # 2x RGBA vs 1x grayscale
+            num_in_ch = 1
+        else:
+            # 4x RGBA vs 2x grayscale
+            num_in_ch = 4
+
+    if num_in_ch * 1 == combined:
+        scale = 4
```

Comment:
This arch uses upconv just like esrgan. Surely you can get the scale based on the number of upconv layers instead of whatever you're doing here. 

For a complex example, take a look at the esrgan code. I didn't write that so I have no idea how it works. But my rust esrgan has simpler code for detecting this.

Reply:
https://github.com/chaiNNer-org/spandrel/blob/bbc371dac1799d654f64622b95dd9d457810ccf1/src/spandrel/architectures/RealESRGAN/arch/RRDBNet.py#L92

Reply:
As you can see, the official realesrgan code has this. We just handle it better in the regular esrgan code. It absolutely is possible to detect this